wgengine/magicsock/magicsock.go

```diff
@@ -172,9 +172,10 @@ func (m *peerMap) forEachEndpoint(f func(ep *endpoint)) {
 	}
 }
 
-// forEachEndpointWithDiscoKey invokes f on every endpoint in m
-// that has the provided DiscoKey.
-func (m *peerMap) forEachEndpointWithDiscoKey(dk key.DiscoPublic, f func(ep *endpoint)) {
+// forEachEndpointWithDiscoKey invokes f on every endpoint in m that has the
+// provided DiscoKey until f returns true or there are no endpoints left to
+// iterate.
+func (m *peerMap) forEachEndpointWithDiscoKey(dk key.DiscoPublic, f func(ep *endpoint) bool) {
```

Comment:
I'd write the signature with named result like `f func(*endpoint) (stop bool)` and omitting the `ep` name.

But I think that's a little backwards than typical. Normally I think it's `(keepGoing bool)`.

But as long as it's named.

Reply:
changed return to `(keepGoing bool)`, updated doc, and omitted `ep`